pyproject.toml

```diff
@@ -0,0 +1,61 @@
+[build-system]
+requires = [
+    "hatchling>=0.11.2",
+    "setuptools<61",
+]
+build-backend = "hatchling.build"
+
+[project]
+name = "test-checks-downloader"
+description = "The test Checks Downloader"
+readme = "README.md"
+license = { text = "BSD-3-Clause" }
+keywords = [
+    "test",
+    "test agent",
+    "test check",
+]
+authors = [
+    { name = "test", email = "test@test.com" },
+]
+classifiers = [
+    "Private :: Do Not Upload",
+    "Development Status :: 5 - Production/Stable",
+    "Intended Audience :: Developers",
+    "Intended Audience :: System Administrators",
+    "License :: OSI Approved :: BSD License",
+    "Programming Language :: Python :: 2.7",
+    "Programming Language :: Python :: 3.8",
+    "Topic :: System :: Monitoring",
+]
+dynamic = [
+    "version",
+]
+
+dependencies = [
+    "Werkzeug==2.3.6",
+    "Jinja2<3.1.2",
+    "itsdangerous>2.1.2",
+    "click>=8.1.3",
+    "blinker<=1.6.2",
+    "importlib-metadata>=3.6.0; python_version < '3.10'",
+    "importlib-metadata==3.8.4; python_version >= '3.10'",
+    "piccolo<1.1.1",
+]
+
+[project.optional-dependencies]
+deps = [
+    "in-toto==1.0.1",
```

Comment:
:exclamation:**Cycode**: Security vulnerability found in newly introduced dependency. 

||| 
 |---|---|
|**Severity**|Medium|
|**Issue**|in-toto: PGP trust model not (fully) considered: GHSA-jjgp-whrp-gq8m|
|**Ecosystem**|PyPI|
|**Dependency**|`in-toto`|
|**Dependency Paths**|`in-toto 1.0.1`|
|**Direct Dependency**|Yes|
|**Upgrade**|``|

 
 ### Impact

This security advisory lists multiple concerns about how in-toto uses PGP keys. The findings are aggregated here, because they are all eligible to the same mitigation strategy. Note that the findings are rated with different severities (see inline) and the highest score was chosen for this advisory:

- **PGP Key Creation Time Not Validated** (severity: low)
  in-toto does not check, if the validity period of a PGP Key (starting with the key creation time) is in the future, when copying the key from GnuPG to a layout, or when verifying signatures. A validity period in the future is usually a sign of a wrong system clock, meaning it can’t be trusted for verifying the validity period. A MITM attacker who is able to manipulate delivered software products might also be able to control the system time by manipulating NTP. In a scenario where an attacker gained control over two expired subkeys with no overlapping validity period, the attacker could set the system time to a time before the validity period of either key, resulting in both keys being accepted.

- **PGP Key Revocation Not Considered** (severity: medium)
 in-toto does not check PGP revocation signatures, when copying the key from GnuPG to a layout, or when verifying signatures. This means that a key may still be accepted in signatures, even if it has been revoked in GnuPG.

- **PGP Key Usage Flags Not Considered** (severity: low)
  in-toto does not check PGP usage flags, when copying the key from GnuPG to a layout, or when verifying signatures. This means that at a key may still be accepted in signatures, even if it is not permitted to sign data as per its key usage flags.


Security auditors recommend to verify these properties at signature verification time.

However, this is not planned, as in-toto does not rely on PGP’s trust model, because it should not be required to consult with a separate PKI/web-of-trust at verification time. Instead the project owner establishes ultimate trust by adding a PGP public key to a layout, and thus is responsible for its validity, and also to revoke the layout, if the key is no longer trusted. The same is true for PGP public keys used to verify a layout.

The preferred mitigation strategy is to verify these properties when exporting a public key from GnuPG, and to clarify usage documentation that no verification against the PGP trust model is performed afterwards.

### References
- "Handle GPG revocation signatures" -- https://github.com/secure-systems-lab/securesystemslib/issues/190
- "inconsistent use of GnuPG" -- https://github.com/in-toto/in-toto/issues/569


 <!-- d0be43c4-b99c-44a1-8092-3b9128b41f04 -->